File: src/main/java/de/marhali/easyi18n/model/SettingsState.java
package de.marhali.easyi18n.model;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Represents the persistent settings which can be configured.
 *
 * @author marhali
 */
public class SettingsState {

    public static final String DEFAULT_PREVIEW_LOCALE = "en";

    private String localesPath;
    private String previewLocale;
    private String keySeparator;

    public SettingsState() {
    }

    public @Nullable String getLocalesPath() {
        return localesPath;
    }

    public void setLocalesPath(String localesPath) {
        this.localesPath = localesPath;
    }

    public @NotNull String getPreviewLocale() {
        return previewLocale != null ? previewLocale : DEFAULT_PREVIEW_LOCALE;
    }

    public void setPreviewLocale(String previewLocale) {
        this.previewLocale = previewLocale;
    }

    public String getKeySeparator() {
        return keySeparator;
    }

    public void setKeySeparator(String keySeparator) {
        this.keySeparator = keySeparator;
    }

    public boolean hasSeparator() {
        return keySeparator != null && keySeparator.length() > 0;
    }
}

File: src/main/kotlin/com/yuukaze/i18next/model/SettingsState.kt
package com.yuukaze.i18next.model

/**
 * Represents the persistent settings which can be configured.
 *
 * @author marhali
 */
class SettingsState {
  var localesPath: String = ""
  var previewLocale: String = ""
    get() = field.ifEmpty { DEFAULT_PREVIEW_LOCALE }
  var keySeparator: String = ""
  var isHasSeparator = false
  var spreadSheetId: String = ""
  var spreadSheetTab: String = ""

  companion object {
    const val DEFAULT_PREVIEW_LOCALE = "en"
  }
}